AudioDeviceExample/AudioDevices/ExampleAVAudioEngineDevice.m

```diff
@@ -339,44 +339,49 @@ - (BOOL)setupPlayoutAudioEngine {
     return YES;
 }
 
-- (void)teardownAudioEngine {
-    [self teardownPlayer];
+- (void)teardownRecordAudioEngine {
+    [_recordEngine stop];
+    _recordEngine = nil;
+}
+
+- (void)teardownPlayoutAudioEngine {
     [_playoutEngine stop];
     _playoutEngine = nil;
+}
 
-    [_recordEngine stop];
-    _recordEngine = nil;
+- (void)teardownAudioEngine {
+    [self teardownFilePlayers];
+
+    [self teardownPlayoutAudioEngine];
+    [self teardownRecordAudioEngine];
 }
 
-- (void)playMusic {
-    NSString *fileName = [NSString stringWithFormat:@"%@/%@", [[NSBundle mainBundle] bundlePath], @"mixLoop.caf"];
-    NSURL *url = [NSURL fileURLWithPath:fileName];
-    AVAudioFile *file = [[AVAudioFile alloc] initForReading:url error:nil];
+- (AVAudioPCMBuffer *)musicBuffer {
+    if (!_musicBuffer) {
```

Comment:
Should we protect against `musicBuffer` being accessed while it's being loaded/prepared in the event of multiple threads attempting to access it? I know our example doesn't do that but in case this is copied and used elsewhere?

Reply:
I noticed this as well, and I am okay either way.

Reply:
I am in an opinion of not protecting unless it is needed. 
In this example code,  it is guaranteed that `self.musicBuffer` is invoked on main thread as a result of user action of a button press to play music.